plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -229,81 +228,80 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize)
         List<ConnectorSplit> splits = new ArrayList<>(maxSize);
         while (splits.size() < maxSize && (fileTasksIterator.hasNext() || fileScanIterator.hasNext())) {
             if (!fileTasksIterator.hasNext()) {
-                FileScanTask wholeFileTask = fileScanIterator.next();
-                if (wholeFileTask.deletes().isEmpty() && noDataColumnsProjected(wholeFileTask)) {
-                    fileTasksIterator = List.of(wholeFileTask).iterator();
+                if (limit.isPresent() && limit.getAsLong() <= outputRowsLowerBound) {
+                    finish();
+                    continue;
```

Comment:
This seems more of a `break` than a `continue` - even though they have both the same effect (due to calling previously `finish()` which empties the `fileScanIterator`